osu.Game/Overlays/BeatmapSet/BeatmapSetBadgePill.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using JetBrains.Annotations;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public class BeatmapSetBadgePill : CircularContainer
```

Comment:
Generally unsure that this much code warrants splitting out. I'd probably rather have two separate pills with a bit of layout code duplicated than starting inheritance trees. It's ended badly in places before (see: tables in rankings overlay).

As it stands this component doesn't even split out the entirety of the shared behaviour (see sprite texts in derived classes specifying the same margins twice).